pkg/declarative/v2/reconciler.go

```diff
@@ -0,0 +1,382 @@
+			Message:            "installation is ready and resources can be used",
+			ObservedGeneration: obj.GetGeneration(),
+		}
+	)
+
+	// Processing
+	if status.State == "" {
+		presetConditions := 2
+		status.Conditions = make([]metav1.Condition, 0, presetConditions)
+		meta.SetStatusCondition(&status.Conditions, resourceCondition)
+		meta.SetStatusCondition(&status.Conditions, installationCondition)
+		status.Synced = make([]Resource, 0)
+		obj.SetStatus(status.WithState(StateProcessing).WithOperation("switched to processing"))
+		return r.ssaStatus(ctx, obj)
+	}
+
+	spec, err := r.Spec(ctx, obj)
+	if err != nil {
+		r.Event(obj, "Warning", "Spec", err.Error())
+		obj.SetStatus(status.WithState(StateError).WithErr(err))
+		return r.ssaStatus(ctx, obj)
+	}
+
+	skr, err := r.initializeClient(ctx, obj, spec)
+	if err != nil {
+		r.Event(obj, "Warning", "ClientInitialization", err.Error())
+		obj.SetStatus(status.WithState(StateError).WithErr(err))
+		return r.ssaStatus(ctx, obj)
+	}
+	resourceConverter := NewResourceConverter(skr, r.Namespace)
+
+	var renderer Renderer
+
+	switch spec.Mode {
+	case RenderModeHelm:
+		renderer = NewHelmRenderer(spec, skr, r.Options)
+		renderer = WrapWithRendererCache(renderer, spec, r.Options)
+	case RenderModeKustomize:
+		renderer = NewKustomizeRenderer(spec, r.Options)
+		renderer = WrapWithRendererCache(renderer, spec, r.Options)
+	case RenderModeRaw:
+		renderer = NewRawRenderer(spec, r.Options)
+	}
+
+	if err := renderer.Initialize(obj); err != nil {
+		return r.ssaStatus(ctx, obj)
+	}
+
+	if err := renderer.EnsurePrerequisites(ctx, obj); err != nil {
+		return r.ssaStatus(ctx, obj)
+	}
+
+	var target, current kube.ResourceList
+
+	if obj.GetDeletionTimestamp().IsZero() {
+		var targetResources *types.ManifestResources
+
+		manifest, err := renderer.Render(ctx, obj)
+		if err != nil {
+			return r.ssaStatus(ctx, obj)
+		}
+
+		targetResources, err = util.ParseManifestStringToObjects(string(manifest))
+		if err != nil {
+			r.Event(obj, "Warning", "ManifestParsing", err.Error())
+			obj.SetStatus(status.WithState(StateError).WithErr(err))
+			return r.ssaStatus(ctx, obj)
+		}
+
+		targetResources.Items = append(targetResources.Items, spec.TargetResources...)
+
+		for _, transform := range r.PostRenderTransforms {
+			if err := transform(ctx, obj, targetResources); err != nil {
+				r.Event(obj, "Warning", "PostRenderTransform", err.Error())
+				obj.SetStatus(status.WithState(StateError).WithErr(err))
+				return r.ssaStatus(ctx, obj)
+			}
+		}
+
+		target, err = resourceConverter.ConvertResourcesFromManifest(targetResources)
+		if err != nil {
+			r.Event(obj, "Warning", "TargetResourceParsing", err.Error())
+			resourceCondition.Status = metav1.ConditionFalse
+			meta.SetStatusCondition(&status.Conditions, resourceCondition)
+			obj.SetStatus(status.WithState(StateError).WithErr(err))
+			return r.ssaStatus(ctx, obj)
+		}
+	} else {
+		target = kube.ResourceList{}
```

Comment:
Maybe it's better to add a comment here to indicate the reason why this target init as empty is because under this condition, `obj` CR is under deletion.